### PR TITLE
Adapt to cairo-lang 0.11.0 

### PR DESCRIPTION
**Pull request by Nathan-SL**

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -10,13 +10,14 @@ dotenv.config();
 const config: HardhatUserConfig = {
     solidity: "0.6.12",
     starknet: {
-        dockerizedVersion: "0.10.3", // alternatively choose one of the two venv options below
+        dockerizedVersion: "0.11.0.1", // alternatively choose one of the two venv options below
```
Comment (FabijanC) [inline hardhat.config.ts:13]: Can we use the latest?

File: contracts/cairo1/cairo1.cairo
Comment (FabijanC) [inline contracts/cairo1/cairo1.cairo:1]: Don't we have some tests which test compiling all contracts in the contracts directory? How are those tests affected by having a cairo1 subdirectory?

File: hardhat.config.ts [hunk 2]
```diff
@@ -10,13 +10,14 @@ dotenv.config();
...
         // uses (my-venv) defined by `python -m venv path/to/my-venv`
         // venv: "path/to/my-venv",
 
         // uses the currently active Python environment (hopefully with available Starknet commands!)
         // venv: "active",
         recompile: false,
+        manifestPath: "path/to/Cargo.toml",
```
Comment (FabijanC) [inline hardhat.config.ts:20]: Can this be commented? I've had the practice to leave uncommented those lines which are generally expected to work. And this one isn't since this path doesn't generally exist on users' computers. Compare with the commented `venv: "path/to/my-venv"`.

File: test/argent-account-test.ts [hunk 3]
```diff
@@ -107,14 +107,18 @@ describe("Argent account", function () {
         const { res: initialBalance } = await mainContract.call("get_balance");
 
         const newGuardianPrivateKey = "0x123";
-        await account.setGuardian(newGuardianPrivateKey);
+        await account.setGuardian(newGuardianPrivateKey, {
+            maxFee: 1e18
```
Comment (FabijanC) [inline test/argent-account-test.ts:111]: Why is this necessary all of a sudden?

File: test/devnet-create-block.test.ts [hunk 4]
```diff
@@ -10,11 +10,13 @@ describe("Devnet create block", function () {
         const latestBlock = await starknet.getBlock();
 
         const emptyBlock1 = await starknet.devnet.createBlock();
-        expect(emptyBlock1.block_number).to.deep.equal(latestBlock.block_number + 1);
-        expect(emptyBlock1.transactions).to.be.empty;
+        expect(Number(emptyBlock1.block_hash)).to.deep.equal(Number(latestBlock.block_hash) + 1);
```
Comment (FabijanC) [inline test/devnet-create-block.test.ts:13]: Why replace block_number with block_hash? And why does this block hash equation hold?

We might be better off with requesting block creation, then getting the block and just asserting that it's empty - i.e. not holding any txs). I.e. what we already have, but remove the hash equality assertion.

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: contracts/cairo1/cairo1.cairo
Comment (Nathan-SL) [reply contracts/cairo1/cairo1.cairo:1]: I've added a condition to skip cairo1 contracts for `npx hardhat starknet-compile`

File: test/argent-account-test.ts (hunk 3 above)
Comment (Nathan-SL) [reply test/argent-account-test.ts:111]: setGuardian doesn't have an implicit maxFee calculation and error `message":"max_fee == 0 is not supported` will be thrown if it's not passed with starknet v0.11.0

File: hardhat.config.ts (hunk 2 above)
Comment (Nathan-SL) [reply hardhat.config.ts]: https://github.com/Shard-Labs/starknet-hardhat-example/pull/104/commits/cce42e19f4fdd4be150246cc056180a8f603bb53

File: test/devnet-create-block.test.ts (hunk 4 above)
Comment (Nathan-SL) [reply test/devnet-create-block.test.ts]: https://github.com/Shard-Labs/starknet-hardhat-example/pull/104/commits/8d02a690f9f40afe9033bdbf699e82de2789c5a9

File: hardhat.config.ts (hunk 1 above)
Comment (FabijanC) [reply hardhat.config.ts:13]: ok

**Review by FabijanC — APPROVED** [review]
Approving for now, but have to see the plugin PR
